packages/near-membrane-base/src/membrane.ts

```diff
@@ -652,6 +682,7 @@ export function init(
             this.preventExtensions = BoundaryProxyHandler.dynamicPreventExtensionsTrap;
             // @ts-ignore
             this.defineProperty = BoundaryProxyHandler.dynamicDefinePropertyTrap;
+            this.targetMarkAsMagic = this.isTargetMarkAsMagic();
```

Comment:
I believe this is not sufficient. This value is set only here, but the problem is that from this logic is only executed if `trapMutations` option is `true`. Which is only in red land. Blue land doesn't use that flag, meaning from blue to red, interactions that depends on the magic mark are not going to work. Now, does it matter? probably a very edge case. The reason why this logic is not used in those scenarios is because the proxies are always dynamic, which is resolved during the static definition of the default traps:

```
        private static defaultSetTrap = trapMutations
            ? BoundaryProxyHandler.pendingSetTrap
            : BoundaryProxyHandler.dynamicSetTrap;
```

We need to figure how to honor that case as well to compute the value.